cpp/arcticdb/storage/memory/memory_storage.cpp

```diff
@@ -58,7 +58,10 @@ namespace arcticdb::storage::memory {
             for (auto &kv : group.values()) {
                 auto it = key_vec.find(kv.variant_key());
 
-                util::check_rte(opts.upsert_ || it != key_vec.end(), "update called with upsert=false but key does not exist");
+                if (!opts.upsert_ && it == key_vec.end()) {
+                    std::string err_message = fmt::format("update called with upsert=false but key does not exist: {}", kv.variant_key());
```

Comment:
Unify with the LMDB message per https://github.com/man-group/ArcticDB/pull/584/files#r1283403586 ?